fhir-search/src/main/java/com/ibm/fhir/search/parameters/ExtensionSearchParametersResourceProvider.java

```diff
@@ -0,0 +1,117 @@
+     * extension-search-parameters.json files) will be stored under their respective tenant-ids as well. The objects
+     * stored in our cache are of type CachedObjectHolder, with each one containing a List<SearchParameter>.
+     */
+    private static TenantSpecificSearchParameterCache searchParameterCache = new TenantSpecificSearchParameterCache();
+
+    @Override
+    protected List<FHIRRegistryResource> getRegistryResources(Class<? extends Resource> resourceType, String url) {
+        if (resourceType != SearchParameter.class || url == null) {
+            return Collections.emptyList();
+        }
+
+        String tenantId = FHIRRequestContext.get().getTenantId();
+        return getTenantOrDefaultExtensionSPs(tenantId).stream()
+                .filter(sp -> url.equals(sp.getUrl().getValue()))
+                .map(sp -> FHIRRegistryResource.from(sp))
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Collection<FHIRRegistryResource> getRegistryResources() {
+        String tenantId = FHIRRequestContext.get().getTenantId();
+        return getTenantOrDefaultExtensionSPs(tenantId).stream()
+                .map(sp -> FHIRRegistryResource.from(sp))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Collection<FHIRRegistryResource> getRegistryResources(Class<? extends Resource> resourceType) {
+        if (resourceType != SearchParameter.class) {
+            return Collections.emptyList();
+        }
+
+        return getRegistryResources();
+    }
+
+    @Override
+    public Collection<FHIRRegistryResource> getProfileResources(String type) {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Collection<FHIRRegistryResource> getSearchParameterResources(String type) {
+        Objects.requireNonNull(type);
+
+        return getRegistryResources().stream()
+                .filter(rr -> type.equals(rr.getType()))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the extended search parameters for the specified tenant,
+     * the default tenant (if no tenant-specific extension search parameter file exists),
+     * or an empty list (if there are no extended search parameters or we hit an error while trying to get them).
+     *
+     * @param tenantId
+     *     the tenant whose SearchParameters should be returned.
+     * @return a list of SearchParameter objects
+     */
+    private static List<SearchParameter> getTenantOrDefaultExtensionSPs(String tenantId) {
+        List<SearchParameter> cachedObjectForTenant = null;
+
+        try {
+            cachedObjectForTenant = searchParameterCache.getCachedObjectForTenant(tenantId);
+
+            if (cachedObjectForTenant == null) {
+
+                if (log.isLoggable(Level.FINE)) {
+                    log.fine(String.format(NO_TENANT_SP_MAP_LOGGING, tenantId, FHIRConfiguration.DEFAULT_TENANT_ID));
```

Comment:
This is now one time? maybe just log out as INFO?

Reply:
it is still called multiple times if extension-search-parameters.json changes and someone is retrieving all registry resources from the registry.  its just that our fhir-search module will only call that on startup now.